packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -752,4 +754,132 @@ describe("<FilterBar />", () => {
       })
     })
   })
+
+  describe("Context use cases", () => {
+    type Items = Array<{ value: string; label: string }>
+
+    type AsyncValues = {
+      department: string[]
+      manager: string[]
+    }
+
+    const MockFilterAsyncComponent = ({
+      id,
+      fetcher,
+    }: {
+      id: string
+      fetcher: (args: Partial<AsyncValues>) => Promise<Items>
+    }): JSX.Element => {
+      const [items, setItems] = useState<Items>([])
+      const { getActiveFilterValues } = useFilterBarContext()
+      const activeFilterVals = getActiveFilterValues()
+
+      useEffect(() => {
+        fetcher(activeFilterVals).then(fetchedItems => {
+          if (JSON.stringify(fetchedItems) !== JSON.stringify(items)) {
+            setItems(fetchedItems)
+          }
+        })
+      }, [JSON.stringify(activeFilterVals)])
+
+      return (
+        <FilterBar.MultiSelect id={id} items={items}>
+          {() => (
+            <FilterMultiSelect.ListBox>
+              {({ allItems }) => (
+                <FilterMultiSelect.ListBoxSection
+                  items={allItems}
+                  sectionName="All Items"
+                >
+                  {item => (
+                    <FilterMultiSelect.Option key={item.key} item={item} />
+                  )}
+                </FilterMultiSelect.ListBoxSection>
+              )}
+            </FilterMultiSelect.ListBox>
+          )}
+        </FilterBar.MultiSelect>
+      )
+    }
+
+    const fetchDepartmentOptions = jest.fn(() =>
+      Promise.resolve([
+        { value: "1", label: "Engineering" },
+        { value: "2", label: "Design" },
+      ])
+    )
+
+    const fetchManagerOptions = jest.fn(() =>
+      Promise.resolve([
+        { value: "3", label: "Superman" },
+        { value: "4", label: "Batman" },
+      ])
+    )
+
+    const config = [
+      {
+        id: "department",
+        name: "Department",
+        Component: (
+          <MockFilterAsyncComponent
+            id="department"
+            fetcher={fetchDepartmentOptions}
+          />
+        ),
+      },
+      {
+        id: "manager",
+        name: "Manager",
+        Component: (
+          <MockFilterAsyncComponent
+            id="manager"
+            fetcher={fetchManagerOptions}
+          />
+        ),
+      },
+    ] satisfies Filters<AsyncValues>
+
+    it("An async component can re-fetch with all active filter values pulled off of the FilterBarContext", async () => {
+      const { getByText, queryByText } = render(
+        <FilterBarWrapper<AsyncValues> filters={config} defaultValues={{}} />
+      )
+
+      // open department filter
+      user.click(getByText("Department"))
```

Comment:
We should always await our user events to simulate real user activity (shown in RTL docs). Also prevents us possibly getting those "did you forget to wrap your test in act" type errors (which can also cause inaccuracies in tests).

Reply:
Fixed.